kornia/geometry/camera/pinhole.py

```diff
@@ -266,6 +277,67 @@ def scale_(self, scale_factor) -> 'PinholeCamera':
         self.width *= scale_factor
         return self
 
+    def project_points(self, point_3d: torch.Tensor) -> torch.Tensor:
+        r"""Project a 3d point in world coordinates onto the 2d camera plane.
+
+        Args:
+            point3d: tensor containing the 3d points to be projected
+                to the camera plane. The shape of the tensor can be :math:`(*, 3)`.
+
+        Returns:
+            tensor of (u, v) cam coordinates with shape :math:`(*, 2)`.
+
+        Example:
+            >>> _ = torch.manual_seed(0)
+            >>> X = torch.rand(1, 3)
+            >>> I = torch.eye(4)[None]
+            >>> E = torch.eye(4)[None]
+            >>> h = torch.ones(1)
+            >>> w = torch.ones(1)
+            >>> pinhole = kornia.geometry.camera.PinholeCamera(K, E, h, w)
+            >>> pinhole.project_points(X)
+            tensor([[5.6088, 8.6827]])
+        """
+        R = self.rotation_matrix
+        t = self.translation_vector
+        point_3d_cam = torch.matmul(R, point_3d.view(3, 1)) + t
+        return project_points(point_3d_cam.view(point_3d_cam.shape[0], 3), self.camera_matrix)
```

Comment:
```suggestion
        return project_points(point_3d_cam.view(-1, 3), self.camera_matrix)
```

Reply:
I changed it to compare results for all batches and number-of-points per batch. Note my revised version